CMakeLists.txt

```diff
@@ -28,6 +28,24 @@ option(use_cppunittest "set use_cppunittest to ON to build CppUnitTest tests on
 option(run_traceability "run traceability tool (default is ON)" ON)
 option(run_reals_check "set run_reals_check to ON to run reals check (default is OFF)." OFF)
 
+if((NOT "${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo") AND (NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Debug"))
+    message(FATAL_ERROR "CMAKE_BUILD_TYPE must be specified and be precisely one of Debug or RelWithDebInfo")
+endif()
+
+# This is done because jemalloc is obtained from vcpkg and brought in with PkgConfig
+# PkgConfig does not support defining different variables with the libraries to be linked for different configurations
+# This forces us to only be able to generate one build configuration (either Debug or RelWithDebInfo)
+# Setting CMAKE_CONFIGURATION_TYPES is the way to limit what configurations are generated
+# so that the user gets only one configuration as option in the Visual Studio GUI
+set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE}" CACHE STRING "" FORCE)

```

Comment:
Should these checks be under "if ((WIN32) AND (${GBALLOC_LL_TYPE} STREQUAL "JEMALLOC"))" because they are only needed for jemalloc?

I guess not a big deal other than changing how we work locally to have to re-run cmake to switch release/debug instead of using the UI.

Reply:
Well, let's see if it gets to be annoying or not :-).